src/opt/ImproveMemoryOps.cpp

```diff
@@ -573,9 +573,13 @@ static bool lOffsets32BitSafe(llvm::Value **variableOffsetPtr, llvm::Value **con
 
     if (variableOffset->getType() != LLVMTypes::Int32VectorType) {
         llvm::SExtInst *sext = llvm::dyn_cast<llvm::SExtInst>(variableOffset);
+        llvm::ZExtInst *zext = llvm::dyn_cast<llvm::ZExtInst>(variableOffset);
         if (sext != nullptr && sext->getOperand(0)->getType() == LLVMTypes::Int32VectorType)
             // sext of a 32-bit vector -> the 32-bit vector is good
             variableOffset = sext->getOperand(0);
+        else if (zext != nullptr && zext->getOperand(0)->getType() == LLVMTypes::Int32VectorType)
+            // zext of a 32-bit vector -> the 32-bit vector is good
+            variableOffset = zext->getOperand(0);
```

Comment:
How about combining the logic?
```suggestion
    if (auto* castInst = llvm::dyn_cast<llvm::CastInst>(variableOffset)) {
        if (llvm::isa<llvm::SExtInst>(castInst) || llvm::isa<llvm::ZExtInst>(castInst)) {
            llvm::Value* operand = castInst->getOperand(0);
            if (operand->getType() == LLVMTypes::Int32VectorType) {
                variableOffset= operand; 
            }
        }
    }
```

Reply:
Fixed